sdk/typescript/src/index.guard.ts

```diff
@@ -726,3 +734,172 @@ export function isTransactionResponse(obj: any, _argumentName?: string): obj is
+    return (
+        (obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+        (obj.Publish !== null &&
+            typeof obj.Publish === "object" ||
+            typeof obj.Publish === "function") &&
+        (obj.Publish.modules !== null &&
+            typeof obj.Publish.modules === "object" ||
+            typeof obj.Publish.modules === "function") &&
+        typeof obj.Publish.modules["__@iterator"] === "function"
+    )
+}
+
+export function isObjectArg(obj: any, _argumentName?: string): obj is ObjectArg {
+    return (
+        ((obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+            isSuiObjectRef(obj.ImmOrOwned) as boolean ||
+            (obj !== null &&
+                typeof obj === "object" ||
+                typeof obj === "function") &&
+            isTransactionDigest(obj.Shared) as boolean)
+    )
+}
+
+export function isCallArg(obj: any, _argumentName?: string): obj is CallArg {
+    return (
+        ((obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+            (obj.Pure !== null &&
+                typeof obj.Pure === "object" ||
+                typeof obj.Pure === "function") &&
+            typeof obj.Pure["__@iterator"] === "function" ||
+            (obj !== null &&
+                typeof obj === "object" ||
+                typeof obj === "function") &&
+            isObjectArg(obj.Object) as boolean)
+    )
+}
+
+export function isStructTag(obj: any, _argumentName?: string): obj is StructTag {
+    return (
+        (obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+        isTransactionDigest(obj.address) as boolean &&
+        isTransactionDigest(obj.module) as boolean &&
+        isTransactionDigest(obj.name) as boolean &&
+        Array.isArray(obj.typeParams) &&
+        obj.typeParams.every((e: any) =>
+            isTypeTag(e) as boolean
+        )
+    )
+}
+
+export function isTypeTag(obj: any, _argumentName?: string): obj is TypeTag {
+    return (
+        ((obj !== null &&
+            typeof obj === "object" ||
+            typeof obj === "function") &&
+            obj.bool === null ||
+            (obj !== null &&
+                typeof obj === "object" ||
+                typeof obj === "function") &&
+            obj.u8 === null ||
+            (obj !== null &&
+                typeof obj === "object" ||
+                typeof obj === "function") &&
+            obj.u64 === null ||
+            (obj !== null &&
+                typeof obj === "object" ||
+                typeof obj === "function") &&
+            obj.u128 === null ||
+            (obj !== null &&
+                typeof obj === "object" ||
+                typeof obj === "function") &&
+            obj.address === null ||
+            (obj !== null &&
```

Comment:
Seems like there's a lot of `object !== null && (typeof obj === "object" || typeof obj === "function")`, could probably make a util function for this.

Reply:
Oh this is an auto-generated file(see the file header)

Reply:
ah my bad, it didn't show up in the github diff

You can mark this as generated so that it never shows diffs in the future: https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github